### PR TITLE
cron setup

### PR DESCRIPTION
**Pull request by excellencetechnologies**

### CODE REVIEW

File: middleware/verify.js [hunk 1]
```diff
@@ -13,6 +13,8 @@ module.exports = function (req, res, next) {
         promise.then(function (verify) {
             URL = verify.get('URL');
             req.URL = URL;
+            Crone_running_time = verify.get('Crone_running_time');
```
Comment (upexcel) [inline middleware/verify.js]: "crone" is wrong.

use "cron" everywhere.

Comment (upexcel) [inline middleware/verify.js]: and explain what is the purpose of this code

File: mods/db.js [hunk 2]
```diff
@@ -1,24 +1,46 @@
...
+    mongoose.connect(config.DB_URL, function (err, db) {
+        var app_url_schema = new Schema({
+            headers: {type: String, required: true, unique: true},
+            url: {type: String, required: true, unique: true},
+            status: {type: String, required: true, unique: true},
+            Cron_running_time: {type: String, required: true, unique: true}
+        });
+        var AppUrls = mongoose.model('AppUrls', app_url_schema);
+            // pattern for crone after 5 min '*/5 * * * *'
+        new CronJob('* * * * * *', function () {
```
Comment (upexcel) [inline mods/db.js]: put cronjob reload code in a new service

File: mods/db.js [hunk 3]
```diff
@@ -1,24 +1,46 @@
...
+            var users = AppUrls;
+            users.findOne({APP_ID: "com.tethr"}, function (err, user) {
```
Comment (upexcel) [inline mods/db.js]: what is users??

File: mods/db.js [hunk 4]
```diff
@@ -1,24 +1,46 @@
...
+                if (err) {
+                    console.log(err)
+                } else if (!user) {
+                    console.log(user)
+                } else {
+                    var Cron_running_time = user.Cron_running_time;
+                    var IST_timezone = 'IST' + moment().tz("Asia/Kolkata").format('h:mm:ss a');
```
Comment (upexcel) [inline mods/db.js]: why is this hard coded?

there 100's of timezone in the world

File: service/cache.js [hunk 5]
```diff
@@ -31,6 +31,15 @@ redisFetch = function (req, productType, id, type, callback) {
...
 redisSet = function (catType, id, limit, optmized_response, type, callback) {
+    if (type == null) {
+        type = '';
+    }
+    if (id == null) {
+        id = '';
+    }
+    if (limit == null) {
+        limit = '';
+    }
```
Comment (upexcel) [inline service/cache.js]: explain reason for this

File: middleware/verify.js (hunk 1 above)
Comment (excellencetechnologies) [inline middleware/verify.js]: there is no need of this . deleted

Comment (excellencetechnologies) [inline middleware/verify.js]: there is no need of this , so deleted

File: mods/db.js (hunk 3 above)
Comment (excellencetechnologies) [inline mods/db.js]: users directly changed to AppUrls

Comment (excellencetechnologies) [inline mods/db.js]: directly chnaged to AppUrls

File: service/cache.js (hunk 5 above)
Comment (excellencetechnologies) [inline service/cache.js]: sir in this if sometimes value come null then it changed to ' '

File: mods/db.js (hunk 2 above)
Comment (excellencetechnologies) [inline mods/db.js]: put all the cron job code in cron.js

File: service/cache.js (hunk 5 above)
Comment (excellencetechnologies) [inline service/cache.js]: why is value coming null? remove this code find out why null is cmoing

File: mods/db.js (hunk 4 above)
Comment (excellencetechnologies) [inline mods/db.js]: this is wrong. we cannot hard code

File: mods/db.js [hunk 6]
```diff
@@ -1,24 +1,29 @@
...
 
 module.exports = function () {
     var mongoose = require('mongoose');
-    mongoose.connect(config.DB_URL);
+    mongoose.connect(config.DB_URL, function (err, db) {
+        var app_url_schema = new Schema({
+            headers: {type: String, required: true, unique: true},
+            url: {type: String, required: true, unique: true},
+            status: {type: String, required: true, unique: true},
+            Cron_running_time: {type: String, required: true, unique: true}
```
Comment (excellencetechnologies) [inline mods/db.js]: C should lowercase

File: mods/db.js [hunk 7]
```diff
@@ -1,24 +1,29 @@
...
+        });
+        var AppUrls = mongoose.model('AppUrls', app_url_schema);
```
Comment (excellencetechnologies) [inline mods/db.js]: make this var app_urls

File: service/cron.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+var CronJob = require('cron').CronJob;
+var moment = require('moment');
+var jstz = require('jstz');
+var timezone = jstz.determine().name();
+
+cron = function (AppUrls) {
+    // pattern for crone  after 5 min '*/5 * * * *'
+    new CronJob('* * * * * *', function () {
+        AppUrls.findOne({APP_ID: "com.tethr"}, function (err, user) {
```
Comment (excellencetechnologies) [inline service/cron.js]: code should work for multiple magento sites not just one

File: service/cron.js [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+            if (err) {
+                console.log(err)
+            } else if (!user) {
+                console.log(user)
+            } else {
+                var Cron_running_time = user.Cron_running_time;
```
Comment (excellencetechnologies) [inline service/cron.js]: C make it small

File: service/cron.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+                var time = moment().tz(timezone).format('h:mm:ss a');
```
Comment (excellencetechnologies) [inline service/cron.js]: this logic doesnt make sense? 
where are converting timezone to server time. have u tested this proper 100% ?

File: mods/db.js (hunk 6 above)
Comment (excellencetechnologies) [inline mods/db.js]: changed to lowercase

File: service/cron.js (hunk 8 above)
Comment (excellencetechnologies) [inline service/cron.js]: done. changed

File: mods/db.js (hunk 7 above)
Comment (excellencetechnologies) [inline mods/db.js]: changed app_urls

File: service/cron.js (hunk 9 above)
Comment (excellencetechnologies) [inline service/cron.js]: make c small

File: service/cron.js (hunk 10 above)
Comment (excellencetechnologies) [inline service/cron.js]: timezone contains the value like asia/calcutta  , so it uses that value and find the current value of asia/calcutta

**Pull request merged**